src/main/java/uk/gov/hmcts/darts/authorisation/component/impl/AuthorisationImpl.java

```diff
@@ -166,6 +168,20 @@ public void authoriseTransformedMediaAgainstUser(Integer transformedMediaId) {
         checkMediaRequestIsRequestedByUser(transformedMediaRepository.getReferenceById(transformedMediaId).getMediaRequest());
     }
 
+    @Override
+    public void authoriseByAnnotationId(Integer annotationId, Set<SecurityRoleEnum> securityRoles) {
+        try {
+            var courthouses = annotationRepository.getReferenceById(annotationId).getHearingList().stream()
```

Comment:
should this not be findById?

Reply:
done, will look into most efficient way as a background task